### PR TITLE
Timer cleanup

### PR DESCRIPTION
**Pull request by toanju**

### CODE REVIEW

File: src/rofl/common/cthread.cpp [hunk 1]
```diff
@@ -298,107 +298,78 @@ cthread::clear_timers()
...
-	timers[timer_id] = new ctimer(timer_id, tspec);
-
+	std::pair<std::set<ctimer>::iterator, bool> rv;
+	bool do_wakeup = false;
 	{
-		bool do_wakeup = false;
-		if (ordered_timers.empty() || (tspec < (*ordered_timers.begin()))) {
+		AcquireReadWriteLock lock(tlock);
+
+		if (ordered_timers.empty() || tspec < ordered_timers.begin()->get_tspec())
```
Comment (DFritzsche) [inline src/rofl/common/cthread.cpp]: This is repeated on line 323.

File: src/rofl/common/cthread.cpp [hunk 2]
```diff
@@ -298,107 +298,78 @@ cthread::clear_timers()
 {
 	AcquireReadWriteLock lock(tlock);
 	ordered_timers.clear();
-	for (auto it : timers)
-		delete it.second;
-	timers.clear();
 };
```
Comment (DFritzsche) [inline src/rofl/common/cthread.cpp]: Just to unify the code, can we get rid of the `;` after the `}` closing the function (and all other functions remaining)?

File: src/rofl/common/ctimer.hpp [hunk 3]
```diff
@@ -66,8 +66,8 @@ class ctimer {
 	 *
 	 */
 	ctimer(
-			uint32_t timer_id,
-			const ctimespec& tspec) :
+			uint32_t timer_id = 0,
+			const ctimespec& tspec = ::timespec{0,0} ) :
```
Comment (DFritzsche) [inline src/rofl/common/ctimer.hpp]: Please consider using the alternative constructor: `const ctimespec& tspec(::timespec {0,0});`

File: src/rofl/common/cthread.cpp [hunk 4]
```diff
@@ -519,31 +490,22 @@ cthread::run_loop()
 			while (true) {
 				/* handle expired timers */
 				std::list<unsigned int> ttypes;
```
Comment (DFritzsche) [inline src/rofl/common/cthread.cpp]: `ttypes` has been made obsolete.

File: src/rofl/common/ctimespec.cpp [hunk 5]
```diff
@@ -118,11 +118,6 @@ ctimespec::operator< (
 		return false;
 	}
 
-	// here: tspec.tv_nsec == t.tspec.tv_nsec
-	if (timer_id < t.timer_id) {
-		return true;
-	}
-
```
Comment (DFritzsche) [inline src/rofl/common/ctimespec.cpp]: we could optimize this and remove the previous `if` (lines 177-119)

File: src/rofl/common/ctimespec.cpp [hunk 6]
```diff
@@ -158,10 +153,6 @@ ctimespec::operator> (
 
 	// here: tspec.tv_nsec == t.tspec.tv_nsec
 
-	if (timer_id > t.timer_id) {
-		return true;
-	}
-
```
Comment (DFritzsche) [inline src/rofl/common/ctimespec.cpp]: we could optimize this and remove the previous `if` (lines 155-157)

File: src/rofl/common/cthread.cpp (hunk 1 above)
Comment (toanju) [inline src/rofl/common/cthread.cpp]: ups, that should have been gone

File: src/rofl/common/cthread.cpp (hunk 2 above)
Comment (toanju) [inline src/rofl/common/cthread.cpp]: it is actually around quite a lot, so I would be happy to remove them all at once. As well adding a clang-format file might be worth it :)

File: src/rofl/common/ctimer.hpp (hunk 3 above)
Comment (toanju) [inline src/rofl/common/ctimer.hpp]: interesting, that this even worked...

File: src/rofl/common/cthread.cpp (hunk 4 above)
Comment (toanju) [inline src/rofl/common/cthread.cpp]: k, I can remove this as well

File: src/rofl/common/ctimespec.cpp (hunk 5 above)
Comment (toanju) [inline src/rofl/common/ctimespec.cpp]: ack

File: src/rofl/common/ctimespec.cpp (hunk 6 above)
Comment (toanju) [inline src/rofl/common/ctimespec.cpp]: ack

**Pull request merged**